openeo/extra/artifacts/_s3sts/sts.py

```diff
@@ -1,5 +1,8 @@
 from __future__ import annotations
 
+import logging
+from random import randint
+from time import sleep
```

Comment:
FYI I usually prefer just to `import time` here and use `time.sleep()` at call time, with these marginal benefits
- minimize number of imported symbols at top level to keep things tidier
- simplifies mocking of `sleep` in testing as it better decouples the tests from how the utilities were imported (e.g. `mock.patch("time.sleep")` would just work, while with `from time import sleep` you would have to worry more about implementation details)